frontend/src/layout/panel-layout/panelLayoutLogic.tsx

```diff
@@ -121,7 +123,42 @@ export const panelLayoutLogic = kea<panelLayoutLogicType>([
                 setPanelWidth: (_, { width }) => width,
             },
         ],
+        panelIsResizing: [
+            false,
+            {
+                setPanelIsResizing: (_, { isResizing }) => isResizing,
+            },
+        ],
+        panelWillHide: [
+            false,
+            {
+                setPanelWillHide: (_, { willHide }) => willHide,
+            },
+        ],
     }),
+    listeners(({ actions, values }) => ({
+        showLayoutPanel: ({ visible }) => {
+            if (visible) {
+                actions.setPanelWillHide(false)
+            }
+        },
+        // We add a visual cue to the panel when it's at or below the minimum width
+        setPanelWidth: ({ width }) => {
+            if (width <= PANEL_LAYOUT_MIN_WIDTH - 1) {
+                actions.setPanelWillHide(true)
+            } else {
+                actions.setPanelWillHide(false)
+            }
+        },
```

Comment:
You could skip these listeners here and modify the data in the reducer directly.

```ts
    reducers({
        panelWillHide: [
            false,
            {
                showLayoutPanel: (state, { visible }) => visible ? false : state
                setPanelWidth: (_, { width }) => width <= PANEL_LAYOUT_MIN_WIDTH - 1
            },
        ],
    })
```

No action needed for me... just thought I'd mention this as an option...